src/main/kotlin/gg/essential/universal/UDesktop.kt

```diff
@@ -71,13 +71,13 @@ object UDesktop {
     private fun openSystemSpecific(file: String): Boolean {
         return when {
             isLinux -> when {
-                isXdg -> runCommand("xdg-open \"$file\"")
-                isKde -> runCommand("kde-open \"$file\"")
-                isGnome -> runCommand("gnome-open \"$file\"")
-                else -> runCommand("kde-open \"$file\"") || runCommand("gnome-open \"$file\"")
+                isXdg -> runCommand("xdg-open", "\"$file\"")
+                isKde -> runCommand("kde-open", "\"$file\"")
+                isGnome -> runCommand("gnome-open", "\"$file\"")
+                else -> runCommand("kde-open", "\"$file\"") || runCommand("gnome-open", "\"$file\"")
             }
-            isMac -> runCommand("open \"$file\"")
-            isWindows -> runCommand("explorer \"$file\"")
+            isMac -> runCommand("bash", "-c", "open \"$file\"")
```

Comment:
This is a semantic change that isn't explained or even mentioned anywhere.
If this needs changing, it must be separate (and explained), otherwise it should be
```suggestion
            isMac -> runCommand("open", file)
```

Reply:
My comment still applies. There's no way anyone will be able to find that explanation in half a year because it's not part of the git history. IMO a change with such a subtle motivation should be documented right there in a comment, otherwise someone might try to clean it up in half a year, test that on 1.8.9, see that it works, and go ahead with the change.

And it should be in a separate commit, not hidden among a bunch of refactors. A few formatting changes in a commit which makes a semantic changes isn't an issue (I'd recommend avoiding it and will point that out, but it doesn't need fixing), the other way around though it's very different, cause then it's very easy to miss the semantic change, and it's difficult to understand which changes were meant to be semantic changes vs which ones were accidental changes.

I'm also a bit skeptical of that original reason given that it doesn't seem likely that the Java headless property would affect `open`, a native Mac program, and given that Minecraft 1.18.2 itself uses `open` directly.